src/Our.Umbraco.Ditto/Extensions/PublishedContentExtensions.cs

```diff
@@ -545,6 +545,13 @@ public static class PublishedContentExtensions
                 // Simple types
                 result = propertyValue;
             }
+            else if (propertyValue is IPublishedContent)
+            {
+                // If the property value is an `IPublishedContent`, then we can use Ditto to map to the target type.
+                result = ((IPublishedContent)propertyValue).As(propertyInfo.PropertyType);
+
+                // TODO: [LK] Could this be also used to handle IEnumerable<IPublishedContent>?
+            }
```

Comment:
Need to add additional check for IEnumerable<IPublishedContent> where to result type is IEnumerable<Entity> and do ditto cast there too